app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,59 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!
+
+  def index
+    @blogs = current_user.blogs
+  end
+
+  def show
+    unless current_user.blogs.include? @blog
```

Comment:
Out of curiosity, why did you make it so that users can only see their own blogs?  I don't believe this is stipulated in the requirements, nor does it reflect how Tumblr does things (as per our email conversation!).

My main concern is that you've put authorization logic here in the show action, but not in the edit or update actions. It's weird that a user can't view other people's blogs but *can* edit them!